app/controllers/topics_controller.rb

```diff
@@ -622,19 +622,20 @@ def bookmark
   def destroy
     topic = Topic.with_deleted.find_by(id: params[:id])
 
-    force_destroy = false
     if params[:force_destroy].present?
       if !guardian.can_permanently_delete?(topic)
         return render_json_error topic.cannot_permanently_delete_reason(current_user), status: 403
       end
-
-      force_destroy = true
     else
       guardian.ensure_can_delete!(topic)
     end
 
-    first_post = topic.posts.with_deleted.order(:post_number).first
-    PostDestroyer.new(current_user, first_post, context: params[:context], force_destroy: force_destroy).destroy
+    PostDestroyer.new(
+      current_user,
+      topic.ordered_posts.with_deleted.first,
+      context: params[:context],
+      force_destroy: params[:force_destroy].present?
```

Comment:
Can `params[:force_destroy]` could be `"false"` here? That would evaluate to true. I wonder if we should use `ActiveRecord::Type::Boolean.new.deserialize` here.

Reply:
Our UI does not ever set force_destroy to false and `ActiveRecord::Type::Boolean.new` has some issues too, but I made the change.